apps/rebar/priv/shell-completion/bash/rebar3

```diff
@@ -2,10 +2,52 @@
 
 _rebar3()
 {
-    local cur prev sopts lopts cmdsnvars
+    local cur prev sopts lopts cmdsnvars profiles rebarconf="rebar.config"
     COMPREPLY=()
-    cur="${COMP_WORDS[COMP_CWORD]}"
-    prev="${COMP_WORDS[COMP_CWORD-1]}"
+    curindex="${COMP_CWORD}"
+    cur="${COMP_WORDS[curindex]}"
+    prev="${COMP_WORDS[curindex-1]}"
+    first="${COMP_WORDS[1]}"
+
+    get_profiles()
+    {
+        if [ -n "${profiles}" ]; then
+            return
+        fi
+
+        local profconfig depth profile namming ch
+
+        if [ -f "$rebarconf" ]; then
+            profconfig=$( cat "$rebarconf" \
+                          | grep -oPz '(?s){profiles.*?(?<=\.)' \
```

Comment:
this can break whenever using periods in content. For example:

```
{profiles, [{
   {test, [{relx, [{sys_config, "./config/test.sys.config"},
   ...
}]}.
```
This will break on the first profile because of the period.

Reply:
Can you try this updated regex?
`(?s){profiles,\s*(\r\n)*\n*\r*\[\s*(\r\n)*\n*\r*.*?(?<=\s*(\r\n)*\n*\r*\}\s*(\r\n)*\n*\r*\.)`

Reply:
```
→ cat rebar.config| grep -oPz '(?s){profiles,\s*(\r\n)*\n*\r*\[\s*(\r\n)*\n*\r*.*?(?<=\s*(\r\n)*\n*\r*\}\s*(\r\n)*\n*\r*\.)'
grep: lookbehind assertion is not fixed length
```

Reply:
I think it's fixed now. Can you try again, @ferd?
```console
$ cat rebar.config | grep -oPz '(?s)(?:{\s*profiles\s*,\s*(\r\n)*\n*\r*\[).*?(?:\s*(\r\n)*\n*\r*}\s*(\r\n)*\n*\r*\.)'
```
I also refactored to a more generic/reusable code.
There is an issue with comments. This regex does not ignore them. I will take a look at this.

Reply:
The comments issue is fixed. Here is the complete command:
```console
cat rebar.config \
| grep -v '\s*%.*' \
| grep -oPz "(?s)(?:{\s*profiles\s*,\s*(\r\n)*\n*\r*\[).*?(?:\s*(\r\n)*\n*\r*}\s*(\r\n)*\n*\r*\.)" \
| tr -d '\0' \
| tr -d '\n' \
| grep -oPz '(?s)(?<=\[).*(?=\])' \
| tr -d '\0' \
| sed -e 's/^[[:space:]]*//'
```